test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -139,6 +138,33 @@ class test_ucp_sockaddr : public ucp_test {
         return UCS_LOG_FUNC_RC_CONTINUE;
     }
 
+    int is_skip_interface(struct ifaddrs *ifa) {
+        int skip = 0;
+
+        if (!has_transport("tcp") && !has_transport("all") &&
+            !ucs::is_rdmacm_netdev(ifa->ifa_name)) {
+            /* IB transports require an IPoIB/RoCE interface since they
+             * use rdmacm for connection establishment, which supports
+             * only IPoIB IP addresses. therefore, if the interface
+             * isn't as such, we continue to the next one. */
+            skip = 1;
+        } else if (!ucs::is_rdmacm_netdev(ifa->ifa_name) &&
+                   !(GetParam().variant & TEST_MODIFIER_CM)) {
+            /* old client-server API (without CM) ran only with
+             * IPoIB/RoCE interface */
+            skip = 1;
+        } else if ((has_transport("tcp") || has_transport("all")) &&
+                   (ifa->ifa_addr->sa_family == AF_INET6)) {
+            /* the tcp transport (and 'all' which may fallback to tcp_sockcmm)
+             * can run either on an rdma-enabled interface (IPoIB/RoCE)
+             * or any interface with IPv4 address because IPv6 isn't supported
+             * by the tcp transport yet */
+            skip = 1;
+        }
+
+        return skip;
+    }
```

Comment:
this should be simplified but since it's gtest and we want to remove it anyway, can merge as is